crates/typst/src/math/attach.rs

```diff
@@ -150,6 +164,44 @@ impl LayoutMath for Packed<PrimesElem> {
     }
 }
 
+/// Stretches a base to accommodate wide top or bottom attachments.
+///
+/// ```example
+/// $ H stretch(=)^"define" U + p V $
+/// $ P => Q stretch(=>, width: #200%) R $
+/// $ V stretch(->, width: #(100% + 1.5em))^("surjection") ZZ $
+/// ```
+#[elem(LayoutMath)]
+pub struct StretchElem {
+    /// The base to stretch.
+    #[required]
+    pub body: Content,
+
+    /// The width of the base to stretch to, relative to the maximum width of
+    /// the base and the attachments (if any).
+    pub width: Smart<Rel<Length>>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct StretchInfo {
+    from: Abs,
+    to: Smart<Rel<Length>>,
+}
+
+impl LayoutMath for Packed<StretchElem> {
+    #[typst_macros::time(name = "math.stretch", span = self.span())]
+    fn layout_math(&self, ctx: &mut MathContext, styles: StyleChain) -> SourceResult<()> {
+        let mut fragment = ctx.layout_into_fragment(self.body(), styles)?;
+        let stretch = StretchInfo { from: fragment.width(), to: self.width(styles) };
+        fragment.set_stretch(Some(stretch));
```

Comment:
Is there any point in adding stretch information to the frame before we check if the fragment can actually be stretched?